### ISSUE TITLE
contrib  diag.mac variable scope

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1454]
Imported from SourceForge on 2024-07-04 16:05:13
Created by **[loupantai](https://sourceforge.net/u/loupantai/)** on 2010-09-21 08:55:40
Original: https://sourceforge.net/p/maxima/bugs/2079

---

I have found a minor \(but annoying\) issue with 

maxima/share/maxima/5.22.1/share/contrib/diag.mac

The definition of the "ghelp" function is working fine unless you 
have already previously defined in your session a variable name "z"

ghelp\(f,labmda,k\):=subst\(labmda,z,diff\(f\(z\),z,k\)/k\!\)$

In other words, the variable "z" should be made internal to the closure

Thanks

Regards,
Jacques

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 16:05:14
Created by **[joanpau](https://sourceforge.net/u/joanpau/)** on 2010-09-23 11:06:23
Original: https://sourceforge.net/p/maxima/bugs/2079/#481f

---

As it is only used for differentiation, I think the best solution is simply to quote z:

ghelp\(f,labmda,k\):=subst\(labmda,'z,diff\(f\('z\),'z,k\)/k\!\)$

BTW: Is labmda misspelled? Should it be lambda?

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 16:05:18
Created by **[loupantai](https://sourceforge.net/u/loupantai/)** on 2010-09-23 13:00:10
Original: https://sourceforge.net/p/maxima/bugs/2079/#d7f2

---

Quoting z is what I did on my own before posting this bug

So, this sounds OK to me as a permanent fix

Thanks

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 16:05:21
Created by **[crategus](https://sourceforge.net/u/crategus/)** on 2010-10-02 18:02:46
Original: https://sourceforge.net/p/maxima/bugs/2079/#8c42

---

As suggested in this thread the argument z of the differentiation has been quoted in revision 1.5 of diag.mac.
Closing this bug report as fixed.
Dieter Kaiser

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 16:05:25
Created by **[crategus](https://sourceforge.net/u/crategus/)** on 2010-10-02 18:02:47
Original: https://sourceforge.net/p/maxima/bugs/2079/#c0c0

---

- **status**: open --> closed
